test_conformance/integer_ops/test_integer_dot_product.cpp

```diff
@@ -0,0 +1,395 @@
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, false, true);
+
+    return result;
+}
+
+template <typename SrcType, typename DstType, size_t N>
+static int test_vectype_packed(cl_device_id deviceID, cl_context context,
+                               cl_command_queue queue, int num_elements)
+{
+    int result = TEST_PASS;
+
+    typedef typename std::make_signed<SrcType>::type SSrcType;
+    typedef typename std::make_signed<DstType>::type SDstType;
+
+    typedef typename std::make_unsigned<SrcType>::type USrcType;
+    typedef typename std::make_unsigned<DstType>::type UDstType;
+
+    // packed dot testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, false);
+
+    // packed dot_acc_sat testing:
+    result |= test_case_dot<UDstType, USrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, SSrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, USrcType, SSrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+    result |= test_case_dot<SDstType, SSrcType, USrcType, N>(
+        deviceID, context, queue, num_elements, true, true);
+
+    return result;
+}
+
+int test_integer_dot_product(cl_device_id deviceID, cl_context context,
+                             cl_command_queue queue, int num_elements)
+{
+    if (is_extension_available(deviceID, "cl_khr_integer_dot_product"))
+    {
+        cl_int error = CL_SUCCESS;
+        int result = TEST_PASS;
+
+        cl_device_integer_dot_product_capabilities_khr dotCaps = 0;
+        error = clGetDeviceInfo(deviceID,
+                                CL_DEVICE_INTEGER_DOT_PRODUCT_CAPABILITIES_KHR,
+                                sizeof(dotCaps), &dotCaps, NULL);
+        test_error(
+            error,
+            "Unable to query CL_DEVICE_INTEGER_DOT_PRODUCT_CAPABILITIES_KHR");
+        test_assert_error(
+            dotCaps & CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_PACKED_KHR,
+            "When cl_khr_integer_dot_product is supported "
+            "CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_PACKED_KHR must be "
+            "supported");
+
+        if (dotCaps
+            & ~(CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_PACKED_KHR
+                | CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_KHR))
+        {
+            log_info("NOTE: found an unknown / untested capability!\n");
```

Comment:
Should this be an error or at least a warning? This might trigger failures with vendor extensions but would also provide an incentive to upstream the required testing.

Reply:
The `log_info` was intended to be more-or-less a warning, yes.  This will show up in any submitted certification logs, I think.

Logging an error seemed like overkill but I'm open to other opinions.

Reply:
I'd personally prefer something stronger than info but I'm OK with the code as is.